src/storage/mvcc/reader/reader.rs

```diff
@@ -673,6 +673,83 @@ mod tests {
         assert!(check_need_gc(Arc::clone(&db), region.clone(), 10, true).is_none());
     }
 
+    #[test]
+    fn test_ts_filter() {
+        let path = tempfile::Builder::new()
+            .prefix("test_ts_filter")
+            .tempdir()
+            .unwrap();
+        let path = path.path().to_str().unwrap();
+        let region = make_region(1, vec![0], vec![13]);
+
+        let db = open_db(path, true);
+        let mut engine = RegionEngine::new(&db, &region);
+
+        engine.put(&[2], 1, 2);
+        engine.put(&[4], 3, 4);
+        engine.flush();
+        engine.put(&[6], 5, 6);
+        engine.put(&[8], 7, 8);
+        engine.flush();
+        engine.put(&[10], 9, 10);
+        engine.put(&[12], 11, 12);
+        engine.flush();
+
+        let snap = RegionSnapshot::from_raw(Arc::clone(&db), region.clone());
+
+        // test set both hint_min_ts and hint_max_ts
+        {
+            let mut iopt = IterOption::default();
+            iopt.set_hint_min_ts(6);
+            iopt.set_hint_max_ts(8);
+            let mut iter = snap.iter_cf(CF_WRITE, iopt).unwrap();
+
+            assert_eq!(iter.seek_to_first(), true);
+            let ts = Key::decode_ts_from(iter.key()).unwrap();
+            assert_eq!(ts.into_inner(), 6);
+
+            assert_eq!(iter.next(), true);
+            let ts = Key::decode_ts_from(iter.key()).unwrap();
+            assert_eq!(ts.into_inner(), 8);
+
+            assert_eq!(iter.next(), false);
+        }
+
+        // test set both hint_min_ts
+        {
+            let mut iopt = IterOption::default();
+            iopt.set_hint_min_ts(10);
+            let mut iter = snap.iter_cf(CF_WRITE, iopt).unwrap();
+
+            assert_eq!(iter.seek_to_first(), true);
+            let ts = Key::decode_ts_from(iter.key()).unwrap();
+            assert_eq!(ts.into_inner(), 10);
+
+            assert_eq!(iter.next(), true);
+            let ts = Key::decode_ts_from(iter.key()).unwrap();
+            assert_eq!(ts.into_inner(), 12);
+
+            assert_eq!(iter.next(), false);
+        }
+
+        // test set both hint_max_ts
+        {
+            let mut iopt = IterOption::default();
+            iopt.set_hint_max_ts(4);
```

Comment:
consider setting it to 7

Reply:
set to 7 by https://github.com/tikv/tikv/pull/6194/commits/334ef77966589669dbaa32397fbeac08d01e6451